python/tskit/trees.py

```diff
@@ -7900,6 +7911,23 @@ def divergence_matrix(
                     mode=mode,
                 )
 
+        if sample_sets is not None:
+            N = len(sample_sets)
+            Dp = np.zeros((D.shape[0], N, N))
+            n = len(samples)
+            count = np.zeros((N, N))
+            for j in range(n):
+                j_sample_set = sample_set_index_map[j]
+                for k in range(n):
+                    k_sample_set = sample_set_index_map[k]
+                    if j_sample_set != k_sample_set:
+                        Dp[:, j_sample_set, k_sample_set] += D[:, j, k]
+                        count[j_sample_set, k_sample_set] += 1
+            # Avoid division by zero
+            count[count == 0] = 1
+            Dp[:, ...] /= count
```

Comment:
Yes, I agree about what to divide by here. But, it'd be clearer (and would get the diagonal entries also) to divide by
```
len(j_sample_set) * len(k_sample_set)
```
in the case that `j_sample_set != k_sample_set`, and by
```
len(j_sample_set) * (len(jsample_set) - 1)
```
for the diagonal.